docs/experienceInDepth.md

```diff
@@ -0,0 +1,265 @@
+    get: 
+      responses:
+        200:
+          description: Ok
+```
+
+If the OpenAPI description does not describe the response payload, then it should be assumed to be of content type `application/octet-stream`.
+
+SDK implementations should return the response payload in the language-native way of providing an untyped set of bytes. This could be a byte-array or some kind of stream response.  
+
+```csharp
+
+   Stream speaker = await apiClient.Speakers.GetAsync();
+
+```
+
+## Response with simple schema
+
+```yaml
+openapi: 3.0.3
+info:
+  title: Response with simple schema
+  version: 1.0.0
+servers:
+  - url: https://example.org/
+paths:
+  /speakers/{speakerId}:
+    get: 
+      parameters:
+        - name: speakerId
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: Ok
+          content:
+            application/json:
+              schema:
+                type: object
+                properties: 
+                  displayName: 
+                    type: string
+```
+
+```csharp
+
+   Speaker speaker = await apiClient.Speakers["23"].GetAsync();
+   string displayName = speaker.DisplayName;
+
+```
+
+## Response with primitive payload
+
+Responses with a content type of `text/plain` should be serialized into a primitive data type.  Unless a Schema object indicates a more precise data type, the payload should be serialized to a string.
+
+```yaml
+openapi: 3.0.3
+info:
+  title: Response with primitive payload
+  version: 1.0.0
+servers:
+  - url: https://example.org/
+paths:
+  /speakers/count:
+    get: 
+      responses:
+        200:
+          description: Ok
+          content:
+            text/plain:
+              schema:
+                type: number
+```
+
+```csharp
+
+   int speakerCount = await apiClient.Speakers.Count;
+
+```
+
+## Filtered collection
+
+```yaml
+openapi: 3.0.3
+info:
+  title: Collection filtered by query parameter
+  version: 1.0.0
+servers:
+  - url: https://example.org/
+paths:
+  /speakers:
+    get: 
+      parameters:
+        - name: location
+          in: query
+          required: false
+          schema:
+            type: string
+      responses:
+        200:
+          description: Ok
+          content:
+            application/json:
+              schema:
+                type: array
+                item:
+                  $ref: "#/components/schemas/speaker"
+components:
+  schemas:
+    speaker:
+      type: object
+      properties: 
+        displayName: 
+          type: string
+        location:
+          type: string
+```
+
+```csharp
+
+   IEnumerable<Speaker> speakers = await apiClient.Speakers.GetAsync(new { Location="Montreal" });
+
+```
+
+## Hetreogenous collection
+
+```yaml
+## Filtered collection
+
+```yaml
+openapi: 3.0.3
+info:
+  title: Collection filtered by query parameter
+  version: 1.0.0
+servers:
+  - url: https://example.org/
+paths:
+  /sessions:
```

Comment:
missing the sessions/{id} path item

Reply:
The example was focused on the filtered collection.  Does adding the item operation add value?

Reply:
@baywet Are you ok if I just put a note on the example "Not implemented yet" ?

Reply:
then this 
```CSharp
Presentation presentation = await apiClient.Sessions["idOfPresentation"].GetAsync() as Presentation;
```
Needs to be updated to 

```CSharp
SessionsResponse response = await apiClient.Sessions.GetAsync();
Presentation presentation = response.Value.First(x => x.Id.Equals("idOfPresentation")) as Presentation;
```

Since the indexer is only introduced by the presence of the additional path item.

Reply:
I'll add the missing path. :-)